filters/multi-color-replace/src/MultiColorReplaceFilter.js

```diff
@@ -0,0 +1,136 @@
+import vertex from './multi-color-replace.vert.js';
+import fragment from './multi-color-replace.frag.js';
+
+/**
+ * MultiColorReplaceFilter
+ *
+ * It's similar to ColorReplaceFilter , but support multi-color.
+ *
+ * @class
+ * @extends PIXI.Filter
+ * @memberof PIXI.filters
+ * @param {Array<number>|Array<Array>} originalColors - The colors that will be changed
+ * @param {Array<number>|Array<Array>} targetColors - The target colors
+ * @param {number} [epsilon=0.05] - Tolerance/sensitivity of the floating-point comparison between colors
+ *                                  (lower = more exact, higher = more inclusive)
+ *
+ * @example
+ *  // replaces true red with true blue, and replaces true green with pure white
+ *  someSprite.filters = [new MultiColorReplaceFilter(
+ *   [0xFF0000, 0x00FF00],
+ *   [0x0000FF, 0xFFFFFF],
+ *   0.001
+ *   )];
+ *  You also could use [R, G, B] as the color
+ *  someOtherSprite.filters = [new MultiColorReplaceFilter(
+ *   [[1,0,0], [0,1,0]],
+ *   [[0,0,1], [1,1,1]],
+ *   0.001
+ *   )];
+ *
+ */
+export default class MultiColorReplaceFilter extends PIXI.Filter
+{
+    constructor(originalColors, targetColors, epsilon = 0.05)
+    {
+        const colorCount = Math.min(originalColors.length, targetColors.length);
+
+        super(
+            vertex,
+            fragment.replace(/%colorCount%/g, colorCount)
```

Comment:
Is it possible to do this without replacing `%colorCount%`? This limits the ability of this class to add or remove additional colors after constructor is called.

Reply:
I would try to pass `colorCount` into shader , but I found that
the  `for loop` in webgl shader doesn't support variable as end-value.
e.g. : 
```
int end = 10;
for (int i =0 ; i < 10 ; i++){
//...
}
```
Will take an error.

Reply:
My suggestion is to set the colorCount to the maximum number of allowed color replacements, for instance, 10. Then in the constructor, throw a warning if the number of replacements exceeds 10.

Reply:
@finscn: What do you think about this suggestion? Of auto-filling `originalColors` and `targetColors` with a maximum length? I would like to be able to change colors after construction.